texttospeech/ssml_addresses/tts.py

```diff
@@ -0,0 +1,131 @@
+    # version of the text, formatted according to the SSML commands. This
+    # function stores the synthetic audio to the designated output file.
+
+    # Args:
+    # ssml_text: string of SSML text
+    # outfile: string name of file under which to save audio output
+
+    # Returns:
+    # nothing
+
+    # Checks to make sure ssml_text is not empty
+    if ssml_text is None:
+        print('ssml_to_audio() error. The input text does not exist.')
+        return
+
+    # Instantiates a client
+    client = texttospeech.TextToSpeechClient()
+
+    # Sets the text input to be synthesized
+    synthesis_input = texttospeech.types.SynthesisInput(ssml=ssml_text)
+
+    # Builds the voice request, selects the language code ("en-US") and
+    # the ssml voice gender ("MALE")
+    voice = texttospeech.types.VoiceSelectionParams(
+        language_code='en-US',
+        ssml_gender=texttospeech.enums.SsmlVoiceGender.MALE)
+
+    # Selects the type of audio file to return
+    audio_config = texttospeech.types.AudioConfig(
+        audio_encoding=texttospeech.enums.AudioEncoding.MP3)
+
+    # Performs the text-to-speech request on the text input with the selected
+    # voice parameters and audio file type
+    response = client.synthesize_speech(synthesis_input, voice, audio_config)
+
+    # Writes the synthetic audio to the output file.
+    with open(outfile, 'wb') as out:
+        out.write(response.audio_content)
+        print('Audio content written to file ' + outfile)
+    # [END tts_ssml_address_audio]
+
+
+# [START tts_ssml_address_ssml]
+def text_to_ssml(inputfile):
+    # Generates SSML text from plaintext.
+    # Given an input filename, this function converts the contents of the text
+    # file into a string of formatted SSML text. This function formats the SSML
+    # string so that, when synthesized, the synthetic audio will pause for two
+    # seconds between each line of the text file. This function also handles
+    # special text characters which might interfere with SSML commands.
+
+    # Args:
+    # inputfile: string name of plaintext file
+
+    # Returns:
+    # A string of SSML text based on plaintext input
+
+    # Parse lines of input file
+    try:
+        open_file = open(inputfile, 'r')
+        raw_lines = open_file.readlines()
+
+    # Checks to make sure that the input file exists
+    except IOError:
+        print('text_to_ssml() error. File ' + inputfile + ' does not exist.')
+        return
+
+    # Define SSML timed break between addresses
+    time = '"2s"'
+    brk = '<break time=' + time + '/>'
+
+    # Initialize SSML script
+    ssml = '<speak>'
+
+    # Iterate through lines of file
+    for line in raw_lines:
+
+        # Replace special characters with HTML Ampersand Character Codes
+        # These Codes prevent the API from confusing text with
+        # SSML commands
+        # For example, '<' --> '&lt;'
+        line = cgi.escape(line)
```

Comment:
Your tutorial is focused on address pronunciation. Are we you expecting them to pass in special characters when providing an address to the tutorial or is this just an edge case?

Reply:
'<' and '>' are probably edge cases. '&' is not--a lot of addresses that I tested use this to say corner addresses e.g. '123 5th St & A St'. I didn't use the ampersand as my commented example because I thought it might be confusing given that an ampersand is part of the HTML syntax. Let me know if you'd like me to reword or rework this section.

Reply:
Right, but the API handles '&' just fine, so this line is probably not needed.

Reply:
It won't mess up the API, but it'll mess up the SSML. None of the SSML commands will compile if there's a rogue character. If I pass <speak>example<break time="100s"/>example</speak> to the API, it'll say 'example', wait 100 seconds, and say 'example again. If I pass the API:  <speak>&<break time="100s"/>example</speak>, it will say 'ampersand example' without the 100 second pause.

Reply:
Sorry just realizing you can't read the SSML I included in the comment because it was formatted out. I'll send you the example in chat.